### PR TITLE
Add option to pass params through the key service.

### PR DESCRIPTION
**Pull request by lamara**
Let's you pass in arbitrary query parameters to the get URL.

### CODE REVIEW

File: key-service.js [hunk 1]
```diff
@@ -23,11 +24,13 @@ function factory(
...
+    var params = $httpParamSerializer(options.params)
     if(options.identity) {
-      url = basePath + '?owner=' + encodeURIComponent(options.identity.id);
+      url = basePath +
+        '?owner=' + encodeURIComponent(options.identity.id) + '&' + params;
       identityId = options.identity.id;
     } else {
-      url = basePath;
+      url = basePath + '?' + params;
       identityId = null;
```
Comment (dlongley) [inline key-service.js]: Probably better to use some URL formatting lib/API here vs. string concatenation, but it will do for now.

**Comment by dlongley** [comment]
LGTM, go ahead and merge, @mattcollier.

File: key-service.js [hunk 2]
```diff
@@ -23,11 +24,13 @@ function factory(
     options = options || {};
     var identityId;
     var url;
+    var params = $httpParamSerializer(options.params)
     if(options.identity) {
-      url = basePath + '?owner=' + encodeURIComponent(options.identity.id);
+      url = basePath +
+        '?owner=' + encodeURIComponent(options.identity.id) + '&' + params;
```
Comment (davidlehn) [inline key-service.js]: Should the params part here and below be added conditionally only if params exist?  Otherwise you get trailing & and ?.

Comment (dlongley) [inline key-service.js]: Yeah, we should do that. a URL formatting lib/API would avoid that as well.

**Comment by dlongley** [comment]
@davidlehn noticed one thing we should fix up, then pass back to @mattcollier.

File: key-service.js (hunk 2 above)
Comment (davidlehn) [inline key-service.js]: Logic starts to get tricky but might want to add owner param to the mix here and just do one serialize call. Would need to refactor to not do that below though.

**Comment by lamara** [comment]
Should be good, passing to @mattcollier

File: key-selector.html [hunk 3]
```diff
@@ -28,5 +28,5 @@
 <stackable-modal stackable="$ctrl.showAddModal"
   stackable-closed="!err && ($ctrl.selected=result)"
   br-lazy="$ctrl.showAddModal" br-lazy-id="br-key-selector-add-key-modal">
-  <br-generate-key-pair-modal br-identity="$ctrl.identity"></br-generate-key-pair-modal>
+  <br-generate-key-pair-modal br-identity="$ctrl.identity" br-params="$ctrl.params"></br-generate-key-pair-modal>
```
Comment (mattcollier) [inline key-selector.html]: @dlongley @lamara put this in as a workaround for the cache issue we have in the key service here: https://github.com/digitalbazaar/bedrock-angular-key/blob/master/key-service.js#L43-L47

**Pull request merged**